docs/rules/unicode-bom.md

```diff
@@ -0,0 +1,60 @@
+# Allow or disallow the Unicode Byte Order Mark (BOM) (unicode-bom)
+
+The Unicode Byte Order Mark (BOM) is used to specify whether code units are big
+endian or little endian. That is, whether the most significant or least
+significant bytes come first. UTF-8 does not require a BOM because byte ordering
+does not matter when characters are a single byte. Since UTF-8 is the dominant
+encoding of the web, we make `"never"` the default option.
+
+## Rule Details
+
+If the `"never"` option is used, this rule requires that the file does not begin
+with the unicode BOM character U+FEFF.
+
+## Options
+
+This rule has a string option:
+
+* `"always"` the unicode BOM is always allowed
```

Comment:
I may have misunderstood #5502. What is the correct behaviour? If "always" then the file must have the BOM and if "never" then it must not have the BOM?